packages/checkbox/src/useCheckbox.ts

```diff
@@ -1,23 +1,183 @@
-import { ref, computed, inject } from 'vue'
+import {
+  ref,
+  computed,
+  inject,
+  getCurrentInstance,
+  onMounted,
+  watch,
+} from 'vue'
+import { toTypeString } from '@vue/shared'
+import { UPDATE_MODEL_EVENT } from '@element-plus/utils/constants'
+import { ICheckboxGroupInstance, ICheckboxProps, PartialReturnType } from './checkbox'
 
-export const useCheckbox = () => {
+export const useCheckboxGroup = () => {
   //todo: ELEMENT
   const ELEMENT = null
   const elForm = inject('elForm', {})
   const elFormItem = inject('elFormItem', {}) as any
-  const _checkboxGroup = inject('CheckboxGroup', {}) as any
-  const focus = ref(false)
-  const isGroup = computed(() => _checkboxGroup && _checkboxGroup.name === 'ElCheckboxGroup')
-  const _elFormItemSize = computed(() => {
+  const checkboxGroup = inject<ICheckboxGroupInstance>('CheckboxGroup', {})
+  const isGroup = computed(() => checkboxGroup && checkboxGroup.name === 'ElCheckboxGroup')
+  const elFormItemSize = computed(() => {
     return (elFormItem || {} as any).elFormItemSize
   })
   return {
     isGroup,
-    focus,
-    _checkboxGroup,
+    checkboxGroup,
     elForm,
     ELEMENT,
-    _elFormItemSize,
+    elFormItemSize,
     elFormItem,
   }
 }
+
+const useModel = (props: ICheckboxProps) => {
+  let selfModel = false
+  const { emit } = getCurrentInstance()
+  const { isGroup, checkboxGroup } = useCheckboxGroup()
+  const isLimitExceeded = ref(false)
+  const store = computed(() => checkboxGroup ? checkboxGroup.modelValue?.value : props.modelValue)
+  const model = computed({
+    get() {
+      return isGroup.value
+        ? store.value
+        : props.modelValue ?? selfModel
+    },
+
+    set(val: unknown) {
+      if (isGroup.value && Array.isArray(val)) {
+        isLimitExceeded.value = false
+
+        if (checkboxGroup.min !== undefined && val.length < checkboxGroup.min.value) {
+          isLimitExceeded.value = true
+        }
+        if (checkboxGroup.max !== undefined && val.length > checkboxGroup.max.value) {
+          isLimitExceeded.value = true
+        }
+
+        isLimitExceeded.value === false && checkboxGroup.changeEvent?.(val)
+      } else {
+        emit(UPDATE_MODEL_EVENT, val)
+        selfModel = val as boolean
+      }
+    },
+  })
+
+  return {
+    model,
+    isLimitExceeded,
+  }
+}
+
+const useCheckboxStatus = (props: ICheckboxProps, { model }: PartialReturnType<typeof useModel>) => {
+  const { isGroup, checkboxGroup, elFormItemSize, ELEMENT } = useCheckboxGroup()
+  const focus = ref(false)
+  const size = computed<string|undefined>(() => checkboxGroup.checkboxGroupSize?.value || elFormItemSize.value || (ELEMENT || {}).size)
+  const isChecked = computed(() => {
+    const value = model.value
+    if (toTypeString(value) === '[object Boolean]') {
+      return Boolean(value)
```

Comment:
Don't you think these two lines of code does not make any sense?
When you tries to know if `value` is Boolean or not, when it's already boolean, why would you use `Boolean(val)` on a `boolean` value?
Since whether `value` is `Object<Boolean>` or `Boolean Literal` applying `toString` on `value` always gets you `[object Boolean]`